### PR TITLE
Update reconyx

### PR DESCRIPTION
**Pull request by parappathekappa**
Add reconyx makernote tag mapping file for hc500 hyperfire, ultrafire and hf2 pro cameras.  Ported from https://fossies.org/linux/Image-ExifTool/lib/Image/ExifTool/Reconyx.pm.  Currently untested.

stubbed out 'cryptic_maker_note' method in classes.py since these images don't contain a 'make' tag in the exif data.  Currently it's hard-wired to assume 'reconyx' make, but probably would be better to make it more generic.  Not sure how many cameras have this same issue.

### CODE REVIEW

**Comment by parappathekappa** [comment]
Sorry, not sure how all the other commits got added into this PR.

**Comment by ianare** [comment]
> Sorry, not sure how all the other commits got added into this PR.

No worries I will squash into one commit.

File: exifread/tags/makernote/__init__.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
 Makernote tag definitions.
 """
+
+from . import apple
+from . import canon
+from . import casio
+from . import fujifilm
+from . import nikon
+from . import olympus
+from . import reconyx
```
Comment (ianare) [inline exifread/tags/makernote/__init__.py:11]: This shouldn't be necessary.

Comment (parappathekappa) [reply exifread/tags/makernote/__init__.py]: Updated!

**Comment by parappathekappa** [comment]
Here is the documentation I used to port over the tags.  I may have used some incorrect mapping or indexing since it doesn't seem like the current implementation is working on that test file.  Let me know if you have any suggestions.

https://exiftool.org/TagNames/Reconyx.html

**Comment by parappathekappa** [comment]
This site looks like it is able to parse the MakerNote data for the HC500 test image: http://metapicz.com/

**Comment by parappathekappa** [comment]
If the current state is too rough, or you don't have any obvious suggestions for best approach I'll keep working on it.  Also, if you have any suggestions for how to get the current mapping working for the HC500 test image let me know, otherwise I'll keep digging on that.  I assumed porting the mapping from the exiftool files would work, but it doesn't seem to be working in its current state.  Maybe I omitted an index or offset.  I noticed that there is an Index2 and Index1 in the header of the exiftool spreadsheets that I didn't account for.

**Comment by parappathekappa** [comment]
No worries!  I'll either rebase off develop or just cut a new branch and
PR.  I'll check on Python3 byte handling as well, maybe that's why I'm
having trouble parsing the tags currently.

-Bryan

File: exifread/tags/makernote/reconyx.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# ported from https://fossies.org/linux/Image-ExifTool/lib/Image/ExifTool/Reconyx.pm
+TAGS_HYPERFIRE = {
+    0x00: ('MakerNoteVersion', make_string),  # Sometimes binary
+    0x01: ('FirmwareVersion', ),
+    0x04: ('FirmwareDate', ),
+    0x06: ('TriggerMode', ),
+    0x07: ('Sequence', ),
+    0x09: ('EventNumber', ),
+    0x0b: ('DateTimeOriginal', ),
+    0x12: ('MoonPhase', ),
```
Comment (ianare) [inline exifread/tags/makernote/reconyx.py:13]: It does look like some of your tag definitions are missing sub-definitions, for example for moon phases,

https://metacpan.org/source/EXIFTOOL/Image-ExifTool-12.00/lib/Image/ExifTool/Reconyx.pm#L244

would look something like this (not tested):

```suggestion
    0x12: ('MoonPhase', {
            0: 'New',
            1: 'New Crescent',
            2: 'First Quarter',
            3: 'Waxing Gibbous',
            4: 'Full',
            5: 'Waning Gibbous',
            6: 'Last Quarter',
            7: 'Old Crescent',
        }
    ),
```

**Pull request closed without merging**